apiserver/apiserver.go

```diff
@@ -538,9 +544,33 @@ func (srv *Server) serveConn(wsConn *websocket.Conn, modelUUID string, apiObserv
 
 	conn := rpc.NewConn(codec, apiObserver)
 
-	h, err := srv.newAPIHandler(conn, modelUUID, host)
+	// Note that we don't overwrite modelUUID here because
+	// newAPIHandler treats an empty modelUUID as signifying
+	// the API version used.
+	resolvedModelUUID, err := validateModelUUID(validateArgs{
+		statePool: srv.statePool,
+		modelUUID: modelUUID,
+	})
+	var (
+		st *state.State
+		h  *apiHandler
+	)
+	if err == nil {
+		st, err = srv.statePool.Get(resolvedModelUUID)
+	}
+
+	if err == nil {
```

Comment:
I think this would have been a little cleaner to retain the newAPIHandler() func and have it return st as well has apiHandler.